src/cpp/src/llm_pipeline_static.cpp

```diff
@@ -233,15 +233,26 @@ enum class GenerateHint {
     BEST_PERF
 };
 
+std::string to_string(GenerateHint h) {
+    switch(h) {
+        case GenerateHint::FAST_COMPILE : 
+            return "FAST_COMPILE";
+        case GenerateHint::BEST_PERF : 
+            return "BEST_PERF";
+        default:
+            OPENVINO_THROW("Unsupported value for type GenerateHint provided");        
+    }
+}
+
 GenerateHint str_to_hint(const std::string& str) {
-    if (str == "FAST_COMPILE") {
+    if (str == to_string(GenerateHint::FAST_COMPILE)) {
         return GenerateHint::FAST_COMPILE;
     }
-    if (str == "BEST_PERF") {
+    if (str == to_string(GenerateHint::BEST_PERF)) {
         return GenerateHint::BEST_PERF;
     }
     OPENVINO_THROW("Unsupported \"GENERATE_HINT\" provided: " +
-                   str + ". Please select either \"FAST_COMPILE\" or \"BEST_PERF\".");
+                   str + ". Please select either \"" + to_string(GenerateHint::BEST_PERF) + "\" or \"" + to_string(GenerateHint::FAST_COMPILE) +"\".");
```

Comment:
Just in case - if `operator<<(ostream&)` is provided for the hint type, `OPENVINO_THROW` concatenates the arguments to a string itself so it could be just

```
    OPENVINO_THROW("Unsupported \"GENERATE_HINT\" provided: ", str, ". Please select either \"", GenerateHint::BEST_PERF, "\" or \"", GenerateHint::FAST_COMPILE, "\".");
```